CBCCasper.thy

```diff
@@ -69,7 +38,32 @@ fun justification :: "message \<Rightarrow> state"
   where
     "justification (Message (_, _, s)) = set s"
 
-(* Definition 2.7 *)
+record params =
+  V :: "validator set"
+  W :: weight
+  t :: threshold
+  C :: "consensus_value set"
+  \<epsilon> :: "message set \<Rightarrow> consensus_value set"
+
+record Protocol =
+  parameters :: params
+  \<Sigma> :: "state set"
+  M :: "message set"
+
+locale Protocol =
+  fixes protocol :: "'p Protocol_scheme" (structure)
+  and params
+  defines params_def: "params \<equiv> parameters protocol"
+
+  assumes W_type: "\<And>w. w \<in> range (W params) \<Longrightarrow> w > 0"
+  and threshould_type: "0 \<le> t params" "t params < Sum (W params ` V params)"
+  and estimator_type: "\<And>s. s \<in> \<Sigma> protocol \<Longrightarrow> \<epsilon> params s \<in> Pow (C params) - \<emptyset>"
```

Comment:
The original paper seems wrong here and I think `Pow (C params) - {∅}` is correct.

https://github.com/cbc-casper/cbc-casper-paper/issues/17

Reply:
Oh, thanks! You're right.